tree_rightview

```diff
@@ -0,0 +1,61 @@
+#include<bits/stdc++.h>
+using namespace std;
+
+struct node
```

Comment:
Since you are using C++, consider making a class instead, with member functions instead of standalone functions like this, which is the C style way of doing this. This will create an easier interface that is pointer free for the user.